packages/coinstac-common/test/models/computation/command.js

```diff
@@ -16,113 +18,204 @@ test('model::CommandComputation constructor', t => {
+    '{ "when": "the data comes ',
+    'a knockin", "dont": "worry',
+    ' too much about it" }',
+  ];
 
-test('model::CommandComputation run (basic, from file)', t => {
-  t.plan(1);
-  const comp = new CommandComputation({
-    cwd: __dirname,
-    type: 'cmd',
-    cmd: 'python',
-    args: ['test-command-good.py'],
-  });
-  comp.run()
-  .then((rslt) => {
-    t.deepEqual(rslt, { bar: 'baz' }, 'computation handles file stdout results');
-    t.end();
-  }, t.end);
-});
+  t.plan(3);
 
-test('model::CommandComputation run fail (basic, invalid cmd)', t => {
-  t.plan(1);
-  const comp = new CommandComputation({
-    cwd: __dirname,
-    type: 'cmd',
-    cmd: 'bogus',
-    args: ['bargus.py'],
-  });
-  comp.run()
-  .then(() => t.end('computation should have failed'))
-  .catch(() => {
-    t.ok('handled invalid command');
-    return t.end();
-  });
-});
+  const run1 = CommandComputation.prototype.run.call({
+    cmd,
+    cwd,
+    isVerbose: true,
+  }, opts);
 
-test('model::CommandComputation run fail (basic, invalid syntax)', t => {
-  t.plan(1);
-  const comp = new CommandComputation({
-    cwd: __dirname,
-    type: 'cmd',
-    cmd: 'python',
-    args: ['test-command-bad-syntax.py'],
-  });
-  comp.run()
-  .then(() => t.end('computation should have failed'))
-  .catch(() => {
-    t.ok('handled bad syntax');
-    return t.end();
-  });
+  stderr.forEach(data => command.stderr.emit('data', data));
+  stdout.forEach(data => command.stdout.emit('data', data));
+  command.emit('exit', 0);
+
+  run1
+    .then(response => {
+      t.ok(
+        spawnStub.calledWithExactly(
+          cmd,
+          ['--run', JSON.stringify(opts)],
+          { cwd }
+        ),
+        'spawns with expected params'
+      );
+      t.deepEqual(
+        response,
+        JSON.parse(stdout.join('')),
+        'returns stdout response'
+      );
+
+      const run2 = CommandComputation.prototype.run.call({
+        args,
+        cmd,
+        cwd,
+        isVerbose: false,
+      }, opts);
+      command.emit('exit', 0);
+
+      return run2;
+    })
+    .then(() => {
+      t.deepEqual(
+        spawnStub.lastCall.args[1],
+        args.concat(['--run', JSON.stringify(opts)]),
+        'adds class args property to spawn params'
+      );
+    })
+    .catch(t.end)
+    .then(spawnStub.restore);
 });
 
-test('model::CommandComputation run fail (basic, file throws Exception)', t => {
-  t.plan(1);
-  const comp = new CommandComputation({
-    cwd: __dirname,
-    type: 'cmd',
-    cmd: 'python',
-    args: ['test-command-bad-exception.py'],
-  });
-  comp.run()
-  .then(() => t.end('computation should have failed'))
-  .catch(() => {
-    t.ok('handled exception');
-    return t.end();
+tape('CommandComputation#run verbose logging', t => {
+  const args = ['some', 'silly', 'arguments'];
+  const cmd = 'macho';
+  const command = getStubbedCommand();
+
+  /**
+   * @todo Don't spy on console methods! Figure out how to inject the logger and
```

Comment:
I believe tape itself uses `console.log` and `console.error`. Makes things confusing. We should inject the logger instance, anyhow.